test/e2e/external_inspection_test.go

```diff
@@ -0,0 +1,108 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/util"
+
+	metal3api "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+)
+
+const hardwareDetails = `{"cpu":{"arch":"x86_64","count":2,"flags":["3dnowprefetch","abm","adx","aes","apic","arat","arch_capabilities","avx","avx2","avx_vnni","bmi1","bmi2","clflush","clflushopt","clwb","cmov","constant_tsc","cpuid","cpuid_fault","cx16","cx8","de","ept","ept_ad","erms","f16c","flexpriority","fma","fpu","fsgsbase","fsrm","fxsr","gfni","hypervisor","ibpb","ibrs","ibrs_enhanced","invpcid","lahf_lm","lm","mca","mce","md_clear","mmx","movbe","movdir64b","movdiri","msr","mtrr","nopl","nx","ospke","pae","pat","pclmulqdq","pdpe1gb","pge","pku","pni","popcnt","pse","pse36","rdpid","rdrand","rdseed","rdtscp","rep_good","sep","serialize","sha_ni","smap","smep","ss","ssbd","sse","sse2","sse4_1","sse4_2","ssse3","stibp","syscall","tpr_shadow","tsc","tsc_adjust","tsc_deadline_timer","tsc_known_freq","umip","vaes","vme","vmx","vnmi","vpclmulqdq","vpid","waitpkg","x2apic","xgetbv1","xsave","xsavec","xsaveopt","xsaves","xtopology"],"model":"12th Gen Intel(R) Core(TM) i9-12900H"},"firmware":{"bios":{"date":"04/01/2014","vendor":"SeaBIOS","version":"1.15.0-1"}},"hostname":"localhost.localdomain","nics":[{"ip":"192.168.222.122","mac":"00:60:2f:31:81:01","model":"0x1af4 0x0001","name":"enp1s0","pxe":true},{"ip":"fe80::570a:edf2:a3a7:4eb8%enp1s0","mac":"00:60:2f:31:81:01","model":"0x1af4 0x0001","name":"enp1s0","pxe":true}],"ramMebibytes":4096,"storage":[{"alternateNames":["/dev/vda","/dev/disk/by-path/pci-0000:04:00.0"],"name":"/dev/disk/by-path/pci-0000:04:00.0","rotational":true,"sizeBytes":21474836480,"type":"HDD","vendor":"0x1af4"}],"systemVendor":{"manufacturer":"QEMU","productName":"Standard PC (Q35 + ICH9, 2009)"}}`
```

Comment:
This was extracted from a BMH that I created manually, by running:

```bash
kubectl get bmh test -o jsonpath="{.status.hardware}"
```

Reply:
You mean like this?
```suggestion
const hardwareDetails = `
{
  "cpu": {
    "arch": "x86_64",
    "count": 2,
    "flags": [
      "3dnowprefetch",
      "abm",
      "adx",
      "aes",
      "apic",
      "arat",
      "arch_capabilities",
      "avx",
      "avx2",
      "avx_vnni",
      "bmi1",
      "bmi2",
      "clflush",
      "clflushopt",
      "clwb",
      "cmov",
      "constant_tsc",
      "cpuid",
      "cpuid_fault",
      "cx16",
      "cx8",
      "de",
      "ept",
      "ept_ad",
      "erms",
      "f16c",
      "flexpriority",
      "fma",
      "fpu",
      "fsgsbase",
      "fsrm",
      "fxsr",
      "gfni",
      "hypervisor",
      "ibpb",
      "ibrs",
      "ibrs_enhanced",
      "invpcid",
      "lahf_lm",
      "lm",
      "mca",
      "mce",
      "md_clear",
      "mmx",
      "movbe",
      "movdir64b",
      "movdiri",
      "msr",
      "mtrr",
      "nopl",
      "nx",
      "ospke",
      "pae",
      "pat",
      "pclmulqdq",
      "pdpe1gb",
      "pge",
      "pku",
      "pni",
      "popcnt",
      "pse",
      "pse36",
      "rdpid",
      "rdrand",
      "rdseed",
      "rdtscp",
      "rep_good",
      "sep",
      "serialize",
      "sha_ni",
      "smap",
      "smep",
      "ss",
      "ssbd",
      "sse",
      "sse2",
      "sse4_1",
      "sse4_2",
      "ssse3",
      "stibp",
      "syscall",
      "tpr_shadow",
      "tsc",
      "tsc_adjust",
      "tsc_deadline_timer",
      "tsc_known_freq",
      "umip",
      "vaes",
      "vme",
      "vmx",
      "vnmi",
      "vpclmulqdq",
      "vpid",
      "waitpkg",
      "x2apic",
      "xgetbv1",
      "xsave",
      "xsavec",
      "xsaveopt",
      "xsaves",
      "xtopology"
    ],
    "model": "12th Gen Intel(R) Core(TM) i9-12900H"
  },
  "firmware": {
    "bios": {
      "date": "04/01/2014",
      "vendor": "SeaBIOS",
      "version": "1.15.0-1"
    }
  },
  "hostname": "localhost.localdomain",
  "nics": [
    {
      "ip": "192.168.222.122",
      "mac": "00:60:2f:31:81:01",
      "model": "0x1af4 0x0001",
      "name": "enp1s0",
      "pxe": true
    },
    {
      "ip": "fe80::570a:edf2:a3a7:4eb8%enp1s0",
      "mac": "00:60:2f:31:81:01",
      "model": "0x1af4 0x0001",
      "name": "enp1s0",
      "pxe": true
    }
  ],
  "ramMebibytes": 4096,
  "storage": [
    {
      "alternateNames": [
        "/dev/vda",
        "/dev/disk/by-path/pci-0000:04:00.0"
      ],
      "name": "/dev/disk/by-path/pci-0000:04:00.0",
      "rotational": true,
      "sizeBytes": 21474836480,
      "type": "HDD",
      "vendor": "0x1af4"
    }
  ],
  "systemVendor": {
    "manufacturer": "QEMU",
    "productName": "Standard PC (Q35 + ICH9, 2009)"
  }
}
`
```
:grimacing: 
I'm not really a fan of this but I can accept it.

Reply:
It does get really long. Could it be put into separate file so it isn't so invasive? It would also make it easier to update if needed.

Reply:
You mean keep it formatted like it is now, or revert to single line?
I'm now wondering also if we should put it in a separate file :thinking: 
Perhaps that would be the best